src/libLookup/Lookup.cpp

```diff
@@ -1642,17 +1708,417 @@ bool Lookup::InitMining()
+                                      m_mediator.m_selfPeer.m_listenPortHost,
+                                      sizeof(uint32_t));
+    curr_offset += sizeof(uint32_t);
+
+    return getLookupOfflineMessage;
+}
+
+std::vector<unsigned char> Lookup::ComposeGetLookupOnlineMessage()
+{
+    LOG_MARKER();
+
+    // getLookupOnlineMessage = [Port]
+    vector<unsigned char> getLookupOnlineMessage
+        = {MessageType::LOOKUP, LookupInstructionType::SETLOOKUPONLINE};
+    unsigned int curr_offset = MessageOffset::BODY;
+
+    Serializable::SetNumber<uint32_t>(getLookupOnlineMessage, curr_offset,
+                                      m_mediator.m_selfPeer.m_listenPortHost,
+                                      sizeof(uint32_t));
+    curr_offset += sizeof(uint32_t);
+
+    return getLookupOnlineMessage;
+}
+
+bool Lookup::GetMyLookupOffline()
+{
+    LOG_MARKER();
+
+    // Remove selfPeerInfo from m_lookupNodes
+    auto iter = std::find(m_lookupNodes.begin(), m_lookupNodes.end(),
+                          m_mediator.m_selfPeer);
+    if (iter != m_lookupNodes.end())
+    {
+        m_lookupNodesOffline.push_back(*iter);
+        m_lookupNodes.erase(iter);
+    }
+    else
+    {
+        LOG_GENERAL(WARNING, "My Peer Info is not in m_lookupNodes");
+        return false;
+    }
+
+    SendMessageToLookupNodesSerial(ComposeGetLookupOfflineMessage());
+    return true;
+}
+
+bool Lookup::GetMyLookupOnline()
+{
+    LOG_MARKER();
+
+    auto iter = std::find(m_lookupNodesOffline.begin(),
+                          m_lookupNodesOffline.end(), m_mediator.m_selfPeer);
+    if (iter != m_lookupNodesOffline.end())
+    {
+        SendMessageToLookupNodesSerial(ComposeGetLookupOnlineMessage());
+        m_lookupNodes.push_back(*iter);
+        m_lookupNodesOffline.erase(iter);
+    }
+    else
+    {
+        LOG_GENERAL(WARNING, "My Peer Info is not in m_lookupNodesOffline");
+        return false;
+    }
+    return true;
+}
+
+bool Lookup::RsyncTxBodies()
+{
+    LOG_MARKER();
+    const Peer& p = GetLookupPeerToRsync();
+    string ipAddr = std::string(p.GetPrintableIPAddress());
+    string port = std::to_string(p.m_listenPortHost);
+    string dbNameStr
+        = BlockStorage::GetBlockStorage().GetDBName(BlockStorage::TX_BODY)[0];
+    string cmdStr;
+    if (ipAddr == "127.0.0.1" || ipAddr == "localhost")
+    {
+        string indexStr = port;
+        indexStr.erase(indexStr.begin());
+        cmdStr = "rsync -iraz --size-only ../node_0" + indexStr + "/"
+            + PERSISTENCE_PATH + "/" + dbNameStr + "/* " + PERSISTENCE_PATH
+            + "/" + dbNameStr + "/";
+    }
+    else
+    {
+        cmdStr = "rsync -iraz --size-only -e \"ssh -o "
+                 "StrictHostKeyChecking=no\" ubuntu@"
+            + ipAddr + ":" + REMOTE_TEST_DIR + "/" + PERSISTENCE_PATH + "/"
+            + dbNameStr + "/* " + PERSISTENCE_PATH + "/" + dbNameStr + "/";
+    }
+    LOG_GENERAL(INFO, cmdStr);
+
+    ExecuteCmd(cmdStr);
```

Comment:
This is a dangerous way of writing code. Our code shouldn't execute cmd line. Is there any C++ rsync lib? If this issue can be resolve easily, create a new issue.

Reply:
I searched and didn't find such third party library so far.

Reply:
Created an issue to track this